packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -268,11 +265,11 @@ class ModalPosition {
   final double left;
 }
 
-abstract class ModalRoute<T> extends LocalHistoryRoute<T> {
+abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T> {
```

Comment:
It's not really inheritance in the mixin. When the class is used as a mixin, it only inserts the class itself into the hierarchy, and then verifies that the thing it now inherits from is type-compatible with the thing that the declaration says it should inherit from. Or something like that.